git-remote-ipfs/src/Network/IPFS/Git/RemoteHelper.hs

```diff
@@ -182,7 +190,12 @@ processPush _ localRef remoteRef = do
         refCid <- ipfs $ resolvePath (cidToText root <> "/" <> remoteRef)
         pure $ refCid >>= hush . cidFromText
 
-    unless (Just localRefCid == remoteRefCid) $ go root localRefCid
+    maxConc <- asks $ ipfsMaxConns . envIpfsOptions
+    runConduit $
+           readObject remoteRefCid localRefCid
+        .| Conduit.conduitVector maxConc
+        .| Conduit.mapM_ (\(batch :: Vector (CID, Git.Object SHA1)) ->
+            forConcurrently_ batch $ pushObject root)
```

Comment:
I haven't looked at any specific invocation of `runConduit` here, but I just wanted to put [this blog post](https://www.well-typed.com/blog/2016/09/sharing-conduit/) on your radar if you don't know about it. Definitely using `conduit` is a powerful tool, but as any powerful weapon, needs to be used carefully 😀 

I am not too concerned about our particular use case (would be easy to spot a memory regression now anyway) but more to add this arrow to your quiver if you didn't have it already 😁

Reply:
I know this article, thanks 😃 

I am reasonably confident that there are no CAFs in both conduit pipelines, but perhaps I should stare at some core to be sure.